core/store/orm/orm.go

```diff
@@ -1113,18 +1406,38 @@ func (orm *ORM) ClobberDiskKeyStoreWithDBKeys(keysDir string) error {
 	return merr
 }
 
-func (orm *ORM) countOf(t interface{}) (int, error) {
+func (orm *ORM) CountOf(t interface{}) (int, error) {
+	err := orm.EnsureAdvisoryLock()
+	if err != nil {
+		return 0, err
+	}
 	var count int
-	return count, orm.DB.Model(t).Count(&count).Error
+	return count, orm.db.Model(t).Count(&count).Error
 }
 
 func (orm *ORM) getRecords(collection interface{}, order string, offset, limit int) error {
-	return orm.DB.
+	err := orm.EnsureAdvisoryLock()
+	if err != nil {
+		return err
+	}
+	return orm.db.
 		Set("gorm:auto_preload", true).
 		Order(order).Limit(limit).Offset(offset).
 		Find(collection).Error
 }
 
+func (orm *ORM) RawDB(fn func(*gorm.DB) error) error {
```

Comment:
Two things--
1) The last code snippet isn't viable (unless we write a fairly complex wrapper of gorm that I worry could be very brittle and buggy), because RawDB() calls `.EnsureAdvisoryLock()` which can return an error.  So we would still end up with an extra few lines due to the extra `if err != nil`.

2) My intent with this callback function pattern was to indicate to the developer that the reference to the `*gorm.DB` object should be short-lived, not stored on other structs or passed to other code, etc., to ensure that we're checking for the advisory lock with some frequency.  If you don't think it's necessary to communicate that and dislike the callback style, I can rework this another way.

Reply:
Got it, disregard.